oop-object_interactions-medium-q5 (#686)
canvas-gamification/canvas-gamification-junit-tests: src/oop/object_interactions/medium/q5/QuestionText.md

# oop-object_interactions-medium-q5

- Java question

## Person Class

> The `Person` class currently has the following items:

Reviewer: Please add in the question context.

- Attributes:
    - A `String` called **myName**
    - An `int` called **numEyes**
    - None of these attributes are accessible from other classes.

- Constructors:
    - A constructor that takes a `String` and an `int` to initialize the **myName** and **numEyes** attributes.
    - It is accessible from other classes.

- Methods:
    - Getter and Setter methods for **myName** and **numEyes** attributes
    - These methods are accessible from other classes.

You must add the following items to the class:

- Methods:

>     - A `receiveDonation` method that takes another **Person** object called donor to do a transplant. It should check

Reviewer: Specify the return type for this method.

>       if the **numEyes** in donor is more than zero. If donor has enough take 1 unit of **numEyes** from donor

Reviewer: I think the wording here is also confusing about which way the transplant goes.

# oop-object_interactions-medium-q5

- Java question

## Question Text

You always need to keep track of transplants in a hospital. For that matter, make a **Person** class with the following

> descriptions to be able to keep track of eye transplants.

Reviewer: Can you add some more context to the question?

> animals that share similar traits. Make a class called **Person** with attributes **myName** and **numEyes** that are

Author: The question has people that can have any number of eyes, or liver in that case. I don't think changing that will make it more realistic.

initialized by the constructor. The class has a method called **receiveDonation** from a donor where a body part is
transplanted as long as that body part is available.

## Person Class

The `Person` class currently has the following items:

- Attributes:
    - A `String` called **myName**
    - An `int` called **numEyes**
    - None of these attributes should be accessible from other classes.

- Constructors:

>     - A constructor that take a `String` and an `int` to initialize the **bloodType** and **volume** attributes.

Reviewer 2: fix variable names here

- Methods:
    - Getter and Setter methods for **myName** and **numEyes** attributes
    - These methods are accessible from other classes.

You must add the following items to the class:

- Methods:
    - A `receiveDonation` method that takes another **Person** object called donor to do a transplant and gives an eye

>       to the object calling the method.It should check if the **numEyes** in donor is more than zero. If donor has

Reviewer: ```suggestion
      to the object calling the method. It should check if the **numEyes** in the donor is more than zero. If donor has
```

    - A `receiveDonation` method that take another **Person** object called donor to do a transplant.
    - It should check if the **numEyes** in donor is non-zero.
    - If donor has enough take 1 unit of **numEyes** from donor
      and give it to the **Person**, and then print `Transplant successful`.
    - A `toString` method that return the `String` of information in the format

>       of `myName + " has " + numEyes + " eyes"`

Reviewer 2: use x y formatting

## TestTransfusion Class

Your `TestTransplant` class must have the following items:

- Main Method:

>     - A `main` method which initializes two **Person** named Tom and Jim where Tom has 2 eyes and jim has zero and print

Reviewer 2: ```suggestion
    - A `main` method that initializes two **Person** objects named Tom and Jim, where Tom has 2 eyes and Jim has zero, and prints
```

>       them using **toString()** Tom first, and then do a transplant for Jim from Tom.

Reviewer 2: ```suggestion
      them using **toString()**. Print Tom's information first and Jim's second. Then have Jim recieve a transplant from Tom.
```

>     - Then print the information of the two **Person**, Tom and Jim in that order, with **toString()**.

Reviewer 2: ```suggestion
    - Print the information of Tom and then Jim using **toString()**.
    -Make sure to print every message on a separate line.
```

# oop-object_interactions-medium-q5

- Java question

## Question Text

Transplant surgeons and researchers investigate the possibility of cross-species transplantation between humans and

> animals that share similar traits. Make a class called **Person** with attributes **myName** and **numEyes** that are

Reviewer 2: it feels weird to mention animals when the class is called "Person"

initialized by the constructor. The class has a method called **receiveDonation** from a donor where a body part is

> transplanted as long as that body part is available.

Reviewer 2: get rid of redundant class information

## Person Class

The `Person` class currently has the following items:

- Attributes:
    - A `String` called **myName**
    - An `int` called **numEyes**
    - None of these attributes are accessible from other classes.

- Constructors:
    - A constructor that takes a `String` and an `int` to initialize the **myName** and **numEyes** attributes.
    - It is accessible from other classes.

- Methods:
    - Getter and Setter methods for **myName** and **numEyes** attributes
    - These methods are accessible from other classes.

You must add the following items to the class:

- Methods:
    - A `receiveDonation` method that takes another **Person** object called donor to do a transplant. It should check
      if the **numEyes** in donor is non-zero. If donor has enough take 1 unit of **numEyes** from donor
      and give it to the **Person**, and then print `Transplant successful`.

>     - A `toString` method that return the `String` of information in the format

Reviewer 2: ```suggestion
    - A `toString` method that returns the `String` of information in the format
```

# oop-object_interactions-medium-q5

- Java question

## Question Text

Transplant surgeons and researchers investigate the possibility of cross-species transplantation between humans and
animals that share similar traits. Make a class called **Person** with attributes **myName** and **numEyes** that are
initialized by the constructor. The class has a method called **receiveDonation** from a donor where a body part is
transplanted as long as that body part is available.

## Person Class

The `Person` class currently has the following items:

- Attributes:
    - A `String` called **myName**
    - An `int` called **numEyes**
    - None of these attributes are accessible from other classes.

- Constructors:

>     - A constructor that take a `String` and an `int` to initialize the **myName** and **numEyes** attributes.

Reviewer 2: ```suggestion
    - A constructor that takes a `String` and an `int` to initialize the **myName** and **numEyes** attributes.
```

    - It is accessible from other classes.

- Methods:
    - Getter and Setter methods for **myName** and **numEyes** attributes
    - These methods are accessible from other classes.

You must add the following items to the class:

- Methods:

>     - A `receiveDonation` method that take another **Person** object called donor to do a transplant. It should check if

Reviewer 2: ```suggestion
    - A `receiveDonation` method that takes another **Person** object called donor to do a transplant. It should check if
```

      the **numEyes** in donor is non-zero. If donor has enough take 1 unit of **numEyes** from donor
      and give it to the **Person**, and then print `Transplant successful`.
    - A `toString` method that return the `String` of information in the format
      of `x has y eyes"` where x is the value of **myName** and y is the value of **numEyes**.

## TestTransplant Class

Your `TestTransplant` class must have the following items:

- Main Method:

>     - A `main` method that initializes two **Person** objects named Tom and Jim, where Tom has 2 eyes and Jim has zero,

Reviewer 2: fix line break formatting here
